### ISSUE TITLE
AWS::StepFunctions::StateMachine - (JSONata) false-positive finding `E3601` for `Comment` in `Choice/Choices` state

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4341]
### CloudFormation Lint Version

1.43.3

### What operating system are you using?

Windows 11

### Describe the bug

cfn-lint incorrectly shows the error `[cfn-lint] E3601: Additional properties are not allowed ('Comment' was unexpected)` when using `Comment` in a `Choice/Choices` state:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test

Resources:
  StepFunctionRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole

  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      RoleArn: !GetAtt StepFunctionRole.Arn
      Definition:
        QueryLanguage: JSONata
        StartAt: Choice
        States:
          Choice:
            Type: Choice
            Choices:
              - Comment: "Test" # <------------------------------
                Condition: "{% $exists($states.input) %}"
                Next: Succeed
            Default: Pass
          Pass:
            Type: Pass
            Next: Succeed
          Succeed:
            Type: Succeed
```

### Expected behavior

cfn-lint doesn't give `E3601` when using `Comment` in a `Choice/Choices` state.

### Reproduction template

See above.
